src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,337 @@
+            )
+        )
+
+        if len(valid_teams) != len(teams):
+            raise serializers.ValidationError("Invalid teams")
+
+        return valid_teams
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Organizations"])
+class OrganizationMemberInviteIndexEndpoint(OrganizationEndpoint):
+    # TODO (mifu67): make these PUBLIC once ready
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (MemberInviteAndStaffPermission,)
+    owner = ApiOwner.ENTERPRISE
+
+    def _can_invite_member(
+        self,
+        request: Request,
+        organization: Organization | RpcOrganization | RpcUserOrganizationContext,
+    ) -> bool:
+        scopes = request.access.scopes
+        is_role_above_member = "member:admin" in scopes or "member:write" in scopes
+        if isinstance(organization, RpcUserOrganizationContext):
+            organization = organization.organization
+
+        if is_role_above_member:
+            return True
+        if "member:invite" not in scopes:
+            return False
+        return not organization.flags.disable_member_invite
+
+    def _invite_member(self, request, organization) -> Response:
+        assigned_org_role = request.data.get("orgRole") or organization_roles.get_default().id
+        allowed_roles = get_allowed_org_roles(request, organization, creating_org_invite=True)
+
+        # We allow requests from integration tokens to invite new members as the member role only
+        if not allowed_roles and request.access.is_integration_token:
+            # Error if the assigned role is not a member
+            if assigned_org_role != "member":
+                raise serializers.ValidationError(
+                    "Integration tokens are restricted to inviting new members with the member role only."
+                )
+            allowed_roles = [organization_roles.get("member")]
+
+        serializer = OrganizationMemberInviteRequestSerializer(
+            data=request.data,
+            context={
+                "organization": organization,
+                "allowed_roles": allowed_roles,
+                "allow_retired_roles": not features.has("organizations:team-roles", organization),
+            },
+        )
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.validated_data
+
+        if ratelimits.for_organization_member_invite(
+            organization=organization,
+            email=result["email"],
+            user=request.user,
+            auth=request.auth,
+        ):
+            metrics.incr(
+                "member-invite.attempt",
+                instance="rate_limited",
+                skip_internal=True,
+                sample_rate=1.0,
+            )
+            return Response({"detail": ERR_RATE_LIMITED}, status=429)
```

Comment:
we might want to check for this first so we don't do extra processing if we should be rate limiting